app/managers/performance_metrics_manager/performance_metrics_batcher.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {AppState, Platform, type AppStateStatus} from 'react-native';
+
+import {sendPerformanceReport} from '@actions/remote/performance';
+import DatabaseManager from '@database/manager';
+import {getConfigValue} from '@queries/servers/system';
+import {toMilliseconds} from '@utils/datetime';
+import {logDebug} from '@utils/log';
+
+const MAX_BATCH_SIZE = 100;
+const INTERVAL_TIME = toMilliseconds({seconds: 60});
+
+class Batcher {
+    private started = false;
+    private batch: PerformanceReportMeasure[] = [];
+    private serverUrl: string;
+    private sendTimeout: NodeJS.Timeout | undefined;
+    private lastAppStateIsActive = AppState.currentState === 'active';
+
+    constructor(serverUrl: string) {
+        this.serverUrl = serverUrl;
+        AppState.addEventListener('change', (appState) => this.onAppStateChange(appState));
+    }
+
+    private start() {
+        this.started = true;
+        clearTimeout(this.sendTimeout);
+        this.sendTimeout = setTimeout(() => this.sendBatch(), INTERVAL_TIME);
+    }
+
+    private async sendBatch() {
+        clearTimeout(this.sendTimeout);
+        if (this.batch.length === 0) {
+            return;
+        }
+
+        const toSend = this.getReport();
+
+        // Empty the batch as soon as possible to avoid race conditions
+        this.batch = [];
+        this.started = false;
+
+        const database = DatabaseManager.serverDatabases[this.serverUrl]?.database;
+        if (!database) {
+            return;
+        }
+
+        const clientPerformanceSetting = await getConfigValue(database, 'EnableClientMetrics');
+        if (clientPerformanceSetting !== 'true') {
+            return;
+        }
+
+        await sendPerformanceReport(this.serverUrl, toSend);
+    }
+
+    private getReport(): PerformanceReport {
+        let start = this.batch[0].timestamp;
+        let end = this.batch[0].timestamp;
+        for (const measure of this.batch) {
+            start = Math.min(start, measure.timestamp);
+            end = Math.max(end, measure.timestamp);
+        }
+        if (start === end) {
+            end += 1;
+        }
+
+        return {
+            version: '0.1.0',
+            labels: {
+                platform: Platform.select({ios: 'ios', default: 'android'}),
+                agent: 'rnapp',
```

Comment:
Similar to my comment on the server PR, I don't think we need to set the agent for mobile reports

Reply:
That seems like a good idea. I've wondered about similar on the web-side, but we're much more restricted in what metrics we can collect about the device there